File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/CalcTaxesInCustModeRequest.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

/**
 * Parameter for CustMode/CalcTaxesInCustMode API.
 */
public class CalcTaxesInCustModeRequest {

    @SerializedName("Transactions")
    private List<Transaction> transactions = new ArrayList<>();

    @SerializedName("Adjustments")
    private List<Transaction> adjustments = new ArrayList<>();

    @SerializedName("Exclusions")
    private List<Exclusion> exclusions = new ArrayList<>();

    @SerializedName("ReturnDetail")
    private Boolean returnDetail = null;

    /**
     * Sets the list of transactions to process.
     * @param transactions
     * @return 
     */
    public CalcTaxesInCustModeRequest transactions(List<Transaction> transactions) {
        this.transactions = transactions;
        return this;
    }

    /**
     * Adds a transaction to this request for processing.
     * @param transactionsItem
     * @return 
     */
    public CalcTaxesInCustModeRequest addTransactionsItem(Transaction transactionsItem) {
        this.transactions.add(transactionsItem);
        return this;
    }

    /**
     * List of transactions to process.
     * @return transactions
     */
    public List<Transaction> getTransactions() {
        return transactions;
    }

    /**
     * Sets the list of transactions to process.
     * @param transactions 
     */
    public void setTransactions(List<Transaction> transactions) {
        this.transactions = transactions;
    }

    /**
     * Sets the list of adjustments to process.
     * @param adjustments
     * @return 
     */
    public CalcTaxesInCustModeRequest adjustments(List<Transaction> adjustments) {
        this.adjustments = adjustments;
        return this;
    }

    /**
     * Adds an adjustment to this request for processing.
     * @param adjustmentsItem
     * @return 
     */
    public CalcTaxesInCustModeRequest addAdjustmentsItem(Transaction adjustmentsItem) {
        this.adjustments.add(adjustmentsItem);
        return this;
    }

    /**
     * List of adjustments to process.
     * @return adjustments
     */
    public List<Transaction> getAdjustments() {
        return adjustments;
    }

    /**
     * Sets the list of adjustments to process.
     * @param adjustments 
     */
    public void setAdjustments(List<Transaction> adjustments) {
        this.adjustments = adjustments;
    }

    /**
     * Sets the list of states for tax exclusions.
     * @param exclusions
     * @return 
     */
    public CalcTaxesInCustModeRequest exclusions(List<Exclusion> exclusions) {
        this.exclusions = exclusions;
        return this;
    }

    /**
     * Adds an exclusion to this request.
     * @param exclusionsItem
     * @return 
     */
    public CalcTaxesInCustModeRequest addExclusionsItem(Exclusion exclusionsItem) {
        this.exclusions.add(exclusionsItem);
        return this;
    }

    /**
     * List of states for tax exclusions.
     * @return exclusions
     */
    public List<Exclusion> getExclusions() {
        return exclusions;
    }

    /**
     * Sets the list of states for tax exclusions.
     * @param exclusions 
     */
    public void setExclusions(List<Exclusion> exclusions) {
        this.exclusions = exclusions;
    }

    /**
     * Sets a value indicating if the individual transaction taxes should be 
     * returned in the response.
     * @param returnDetail
     * @return 
     */
    public CalcTaxesInCustModeRequest returnDetail(Boolean returnDetail) {
        this.returnDetail = returnDetail;
        return this;
    }

    /**
     * Indicates if the individual transaction taxes should be returned in the
     * response.
     * @return returnDetail
     */
    public Boolean getReturnDetail() {
        return returnDetail;
    }

    /**
     * Sets a value indicating if the individual transaction taxes should be 
     * returned in the response.
     * @param returnDetail 
     */
    public void setReturnDetail(Boolean returnDetail) {
        this.returnDetail = returnDetail;
    }
}

File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/CalcTaxesInCustModeRequest.js
/**
 * Parameter for CustMode/CalcTaxesInCustMode API.
 */

function CalcTaxesInCustModeRequest() {

    this.Transactions = [];
    this.Adjustments = [];
    this.Exclusions = [];
    this.ReturnDetail = false;

    /**
     * Sets the list of transactions to process.
     * @param transactions Array of Transaction
     * @return Reference to this object for method chaining. 
     */
    this.transactions = function (transactions) {
        this.Transactions = transactions;
        return this;
    }

    /**
     * Adds a transaction to this request for processing.
     * @param transactionsItem Transaction object.
     * @return Reference to this object for method chaining. 
     */
    this.addTransactionsItem = function (transactionsItem) {
        this.Transactions.push(transactionsItem);
        return this;
    }

    /**
     * List of transactions to process.
     * @return Array of Transaction
     */
    this.getTransactions = function () {
        return this.Transactions;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of transactions to process.
     * @param transactions Array of Transaction 
     */
    this.setTransactions = function (transactions) {
        this.Transactions = transactions;
    }

    /**
     * Sets the list of adjustments to process.
     * @param adjustments Array of Transaction
     * @return Reference to this object for method chaining. 
     */
    this.adjustments = function (adjustments) {
        this.Adjustments = adjustments;
        return this;
    }

    /**
     * Adds an adjustment to this request for processing.
     * @param adjustmentsItem Transaction object.
     * @return Reference to this object for method chaining. 
     */
    this.addAdjustmentsItem = function (adjustmentsItem) {
        this.Adjustments.push(adjustmentsItem);
        return this;
    }

    /**
     * List of adjustments to process.
     * @return Array of Transaction
     */
    this.getAdjustments = function () {
        return this.Adjustments;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of adjustments to process.
     * @param adjustments Array of Transaction 
     */
    this.setAdjustments = function (adjustments) {
        this.Adjustments = adjustments;
    }

    /**
     * Sets the list of states for tax exclusions.
     * @param exclusions Array of Exclusion
     * @return Reference to this object for method chaining. 
     */
    this.exclusions = function (exclusions) {
        this.Exclusions = exclusions;
        return this;
    }

    /**
     * Adds an exclusion to this request.
     * @param exclusionsItem Exclusion object.
     * @return Reference to this object for method chaining. 
     */
    this.addExclusionsItem = function (exclusionsItem) {
        this.Exclusions.push(exclusionsItem);
        return this;
    }

    /**
     * List of states for tax exclusions.
     * @return Array of Exclusion
     */
    this.getExclusions = function () {
        return this.Exclusions;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of states for tax exclusions.
     * @param exclusions Array of Exclusion 
     */
    this.setExclusions = function (exclusions) {
        this.Exclusions = exclusions;
    }

    /**
     * Sets a value indicating if the individual transaction taxes should be 
     * returned in the response.
     * @param returnDetail Boolean
     * @return Reference to this object for method chaining. 
     */
    this.returnDetail = function (returnDetail) {
        this.ReturnDetail = returnDetail;
        return this;
    }

    /**
     * Indicates if the individual transaction taxes should be returned in the
     * response.
     * @return Boolean
     */
    this.getReturnDetail = function () {
        return this.ReturnDetail;
    }

    /* Reference to this object for method chaining.*
     * Sets a value indicating if the individual transaction taxes should be 
     * returned in the response.
     * @param returnDetail Boolean 
     */
    this.setReturnDetail = function (returnDetail) {
        this.ReturnDetail = returnDetail;
    }
}
